src/types/signature.rs

```diff
@@ -59,6 +60,26 @@ impl FunctionType {
     pub fn pure(self) -> Signature {
         self.with_input_extensions(ExtensionSet::new())
     }
+
+    pub(crate) fn validate(
```

Comment:
This was just moved out of `types.rs` in e5ec9d64393966930966e0cf7ced6fb03dd10858, the only "new" bit is adding the `type_vars` parameter